mlir/include/mlir/IR/CMakeLists.txt

```diff
@@ -15,48 +15,58 @@ set(LLVM_TARGET_DEFINITIONS BuiltinAttributes.td)
 mlir_tablegen(BuiltinAttributes.h.inc -gen-attrdef-decls)
 mlir_tablegen(BuiltinAttributes.cpp.inc -gen-attrdef-defs)
 add_public_tablegen_target(MLIRBuiltinAttributesIncGen)
+add_dependencies(mlir-generic-headers MLIRBuiltinAttributesIncGen)
 
 set(LLVM_TARGET_DEFINITIONS BuiltinAttributeInterfaces.td)
 mlir_tablegen(BuiltinAttributeInterfaces.h.inc -gen-attr-interface-decls)
 mlir_tablegen(BuiltinAttributeInterfaces.cpp.inc -gen-attr-interface-defs)
 add_public_tablegen_target(MLIRBuiltinAttributeInterfacesIncGen)
+add_dependencies(mlir-generic-headers MLIRBuiltinAttributeInterfacesIncGen)
 
 set(LLVM_TARGET_DEFINITIONS BuiltinDialect.td)
 mlir_tablegen(BuiltinDialect.h.inc -gen-dialect-decls)
 mlir_tablegen(BuiltinDialect.cpp.inc -gen-dialect-defs)
 add_public_tablegen_target(MLIRBuiltinDialectIncGen)
+add_dependencies(mlir-generic-headers MLIRBuiltinDialectIncGen)
 
 set(LLVM_TARGET_DEFINITIONS BuiltinDialectBytecode.td)
 mlir_tablegen(BuiltinDialectBytecode.cpp.inc -gen-bytecode -bytecode-dialect="Builtin")
 add_public_tablegen_target(MLIRBuiltinDialectBytecodeIncGen)
+add_dependencies(mlir-generic-headers MLIRBuiltinDialectBytecodeIncGen)
 
 set(LLVM_TARGET_DEFINITIONS BuiltinLocationAttributes.td)
 mlir_tablegen(BuiltinLocationAttributes.h.inc -gen-attrdef-decls)
 mlir_tablegen(BuiltinLocationAttributes.cpp.inc -gen-attrdef-defs)
 add_public_tablegen_target(MLIRBuiltinLocationAttributesIncGen)
+add_dependencies(mlir-generic-headers MLIRBuiltinLocationAttributesIncGen)
 
 set(LLVM_TARGET_DEFINITIONS BuiltinOps.td)
 mlir_tablegen(BuiltinOps.h.inc -gen-op-decls)
 mlir_tablegen(BuiltinOps.cpp.inc -gen-op-defs)
 add_public_tablegen_target(MLIRBuiltinOpsIncGen)
+add_dependencies(mlir-generic-headers MLIRBuiltinOpsIncGen)
 
 set(LLVM_TARGET_DEFINITIONS BuiltinTypes.td)
 mlir_tablegen(BuiltinTypes.h.inc -gen-typedef-decls)
 mlir_tablegen(BuiltinTypes.cpp.inc -gen-typedef-defs)
 add_public_tablegen_target(MLIRBuiltinTypesIncGen)
+add_dependencies(mlir-generic-headers MLIRBuiltinTypesIncGen)
 mlir_tablegen(BuiltinTypeConstraints.h.inc -gen-type-constraint-decls)
 mlir_tablegen(BuiltinTypeConstraints.cpp.inc -gen-type-constraint-defs)
 add_public_tablegen_target(MLIRBuiltinTypeConstraintsIncGen)
+add_dependencies(mlir-generic-headers MLIRBuiltinTypeConstraintsIncGen)
 
 set(LLVM_TARGET_DEFINITIONS BuiltinTypeInterfaces.td)
 mlir_tablegen(BuiltinTypeInterfaces.h.inc -gen-type-interface-decls)
 mlir_tablegen(BuiltinTypeInterfaces.cpp.inc -gen-type-interface-defs)
 add_public_tablegen_target(MLIRBuiltinTypeInterfacesIncGen)
+add_dependencies(mlir-generic-headers MLIRBuiltinTypeInterfacesIncGen)
 
 set(LLVM_TARGET_DEFINITIONS TensorEncoding.td)
 mlir_tablegen(TensorEncInterfaces.h.inc -gen-attr-interface-decls)
 mlir_tablegen(TensorEncInterfaces.cpp.inc -gen-attr-interface-defs)
 add_public_tablegen_target(MLIRTensorEncodingIncGen)
+add_dependencies(mlir-generic-headers MLIRTensorEncodingIncGen)
```

Comment:
Why a macro for handling `mlir-headers` but not one for `mlir-generic-headers`?

Reply:
The main reason was because I couldn't think of a good name for it. And `mlir-headers` is used far more frequently than `mlir-generic-headers`.

Reply:
What about descriptive names:
- `add_mlir_dialect_tablegen_target`
- `add_mlir_generic_tablegen_target`